apps/application/flow/step_node/image_generate_step_node/impl/base_image_generate_node.py

```diff
@@ -0,0 +1,100 @@
+        for val in chat_record.details.values():
+            if self.node.id == val['node_id'] and 'image_list' in val:
+                if val['dialogue_type'] == 'WORKFLOW':
+                    return chat_record.get_ai_message()
+                return AIMessage(content=val['answer'])
+        return chat_record.get_ai_message()
+
+    def get_history_message(self, history_chat_record, dialogue_number):
+        start_index = len(history_chat_record) - dialogue_number
+        history_message = reduce(lambda x, y: [*x, *y], [
+            [self.generate_history_human_message(history_chat_record[index]),
+             self.generate_history_ai_message(history_chat_record[index])]
+            for index in
+            range(start_index if start_index > 0 else 0, len(history_chat_record))], [])
+        return history_message
+
+    def generate_history_human_message(self, chat_record):
+
+        for data in chat_record.details.values():
+            if self.node.id == data['node_id'] and 'image_list' in data:
+                image_list = data['image_list']
+                if len(image_list) == 0 or data['dialogue_type'] == 'WORKFLOW':
+                    return HumanMessage(content=chat_record.problem_text)
+                return HumanMessage(content=data['question'])
+        return HumanMessage(content=chat_record.problem_text)
+
+    def generate_prompt_question(self, prompt):
+        return self.workflow_manage.generate_prompt(prompt)
+
+    def generate_message_list(self, question: str, history_message):
+        return [
+            *history_message,
+            question
+        ]
+
+    @staticmethod
+    def reset_message_list(message_list: List[BaseMessage], answer_text):
+        result = [{'role': 'user' if isinstance(message, HumanMessage) else 'ai', 'content': message.content} for
+                  message
+                  in
+                  message_list]
+        result.append({'role': 'ai', 'content': answer_text})
+        return result
+
+    def get_details(self, index: int, **kwargs):
+        return {
+            'name': self.node.properties.get('stepName'),
+            "index": index,
+            'run_time': self.context.get('run_time'),
+            'history_message': [{'content': message.content, 'role': message.type} for message in
+                                (self.context.get('history_message') if self.context.get(
+                                    'history_message') is not None else [])],
+            'question': self.context.get('question'),
+            'answer': self.context.get('answer'),
+            'type': self.node.type,
+            'message_tokens': self.context.get('message_tokens'),
+            'answer_tokens': self.context.get('answer_tokens'),
+            'status': self.status,
+            'err_message': self.err_message,
+            'image_list': self.context.get('image_list'),
+            'dialogue_type': self.context.get('dialogue_type')
+        }
```

Comment:
The provided code does not appear to have any significant issues with formatting. However, the docstring appears to be slightly more concise compared to its previous version, making it easier to read while also being consistent across all major versions of Python.

However, there might be some potential improvements that could improve efficiency:

- Instead of using `reductions` directly on list comprehension, you should convert each object in the list into a tuple before reducing, which can avoid unnecessary copying and may lead to faster execution times.
  python
    return reduce(lambda acc, elem: [*acc, *elem], [
        # Convert each element (which is an instance of BaseMessage subclass) to a list/tuple containing just the content (not type)
        [self.generate_history_human_message(chat_record), 
         self.generate_history_ai_message(chat_record)]
       for chat_record in history_chat_record]

But this solution isn't strictly necessary; however, it could help make the API documentation cleaner without having to change much core logic.
  
Note that these suggestions assume familiarity with Python's idiomatic conventions and practices such as handling list comprehensions carefully or converting objects explicitly into tuples when performing reductions over lists. The actual implementation would still need to verify whether passing individual objects through the list comprehension yields better performance than flattening the data first.